agent/command/registry.go

```diff
@@ -200,3 +206,50 @@ func (r *commandRegistry) renderCommands(commandInfo model.PluginCommandConf,
 
 	return out, nil
 }
+
+// BlockInfo contains information about the enclosing block in which a function
+// or standalone command runs. For example, this would contain information about
+// the pre block that contains a particular shell.exec command.
+type BlockInfo struct {
+	// Block is the name of the block that the command is part of.
+	Block string
+	// CmdNum is the ordinal of a command in the block.
+	CmdNum int
+	// TotalCmds is the total number of commands in the block.
+	TotalCmds int
+}
+
+// FunctionInfo contains information about the enclosing function in which a
+// command runs. For example, this would contain information about the second
+// shell.exec that runs in a function.
+type FunctionInfo struct {
+	// Function is the name of the function that the command is part of.
+	Function string
+	// SubCmdNum is the ordinal of the command within the function.
+	SubCmdNum int
+	// TotalSubCmds is the total number of sub-commands within the function.
+	TotalSubCmds int
+}
+
+// GetDefaultDisplayName returns the default display name for a command.
+// cmdNum is command/function number, subCmdNum only applies for subcmds in
+// functions
+func GetDefaultDisplayName(commandName string, blockInfo BlockInfo, funcInfo FunctionInfo) string {
+	displayName := fmt.Sprintf("'%s'", commandName)
+	if funcInfo.Function != "" {
+		displayName = fmt.Sprintf("%s in function '%s'", displayName, funcInfo.Function)
+	}
+	if blockInfo.CmdNum > 0 && blockInfo.TotalCmds > 0 {
+		if funcInfo.SubCmdNum > 0 && funcInfo.TotalSubCmds > 1 {
+			// Include the function sub-command number only if the function runs
+			// more than one command.
+			displayName = fmt.Sprintf("%s (step %d.%d of %d)", displayName, blockInfo.CmdNum, funcInfo.SubCmdNum, blockInfo.TotalCmds)
```

Comment:
I wonder if the "step x of x" part is useful in the UI. Do we maybe want to only add that for the task logs? 

 I also noticed in the testing staging link that the display name is `'subprocess.exec' in function 'run-make' (step 3 of 3)` and doesn't include the block. Is it because it's not in a pre or post block? If that's the case I think it would be helpful to test one that does include the block so that we can see what that would look like.